aptos-move/framework/aptos-framework/sources/code.move

```diff
@@ -200,6 +223,54 @@ module aptos_framework::code {
         }
     }
 
+    /// Check that the upgrade policies of all packages are equal or higher quality than this package.
+    fun check_dependencies(publish_address: address, pack: &PackageMetadata) acquires PackageRegistry {
```

Comment:
My main concern with this is this would make module publishing a lot more costly. Even if these checks are not recursive, just one level of checking should already significantly increase the cost you have to load an entire package from a different account. For example, I want to deploy KevinModule that depends on aptos_framework::stake, this would cost me a ton as it would load the PackageRegistry for the entire aptos_framework.

Reply:
For the core frameworks (0x1, 0x2, ..) its actually not costly because those are exempted from the dependency check, so their registry is never loaded. The exemption is needed because otherwise you could never define an immutable package which uses the frameworks. I agree that long term that is a cost factor. BTW, I would generally tend pay a bit more in computation resources (without charging the user) for the sake of more security.

Reply:
what if someone introduces an arbitrarily expensive contract to lock me into being unable to upgrade my contract or to just make it ridiculously expensive. I wonder if we should enable this to be optional or to leave this as unimplemented and just get the updated data structures in place.

thoughts?